src/Renci.SshNet/ShellStream.cs

```diff
@@ -52,6 +54,19 @@ public bool DataAvailable
                 }
             }
         }
+        /// <summary>
+        /// Sends a Window Change Request via the Channel.
+        /// </summary>
+        /// <param name="columns">New screen width in # of columns</param>
+        /// <param name="rows">New screen height in # of rows</param>
+        /// <param name="width">New screen width in Pixels</param>
+        /// <param name="height">New screen height in Pixels</param>
+        /// <returns>true when change is successful, or false when channel is NOT open or the request </returns>
+        public bool WindowChangeRequest(uint columns, uint rows, uint width, uint height )
+        {
+            if (_channel==null || !_channel.IsOpen) return false;
```

Comment:
Please have it throw an **ObjectDisposedException** when *_channel* is **null**, and an **InvalidOperationException** if the channel is closed. Please document this as such and add corresponding unit tests.

I would've prefered to have the method on **ChannelSession** also throw if the channel were closed, but that would be considered a breaking change. That method currently either throws or returns **true**, never **false**. It also returns **true** if the channel is closed 😌

Reply:
I always ask myself - as library developers, can we KNOW that it is **Exceptional** that a method is called when in the wrong state or is it simply _unexpected_.  Often the answer is that you can't know the answer because we don't know the context of how the method is being used.   The developer using the library will know if the situation is exceptional and can throw if it is.  All we need to know is do is say whether or not it succeeded.

If you throw an exception in a library - it has to be fundamentally wrong to do so.  In this case, I don't think it is this case - it is merely **_unexpected_** and of no consequence.  There wont be any data coming back to be "in the wrong shape" so it wont matter that the method failed.   There might be an argument for Throwing the exception if the shell size had not been set already, but as this happens in the constructor that is not possible.

The scenario that comes to mind is, in the CanClose Event of a Desktop App, OR the Close browser event, the shape of the window may change during clean up.  If that happens, the event to change window shape MAY still be attached causing the event to fire when the channel has already closed.  It is much simpler to handle for the developer if the method does not throw an exception and the unexpected event can be ignored.   With an exception, in order to properly clean up, the developer must remember to unsubscribe to events or risk an exception being raised when it is of no consequence. 

Returning false allows to the developer to decide if failed call to change window size is exceptional.  So I think returning a boolean is the right call here.

Reply:
Yes, change window might be a better name.  Happy to do this.  Should I proceed?

Reply:
Ok changed in commit [61d5324]

Reply:
With regards to an example application - I have a fully worked example in branch `glenkleidon:feature/add_vs_2022_solution` 

Should consider doing a pull request on that one but I need some help with a failed test case.

Reply:
Actually the blocker is the failed test case on https://github.com/sshnet/SSH.NET/pull/1024